### ISSUE TITLE
Title label gets truncated instead of scrolling with custom duration

### ISSUE THREAD

**Issue by egenvall** [Daltron/NotificationBanner#10]
The titlelabel of a NotificationBanner does not scroll if a custom duration is set for the banner, instead it gets truncated at the end.
I didn't check for the subtitle or status banner.

     func successNotificationBanner(_ text: String, subtitle: String){
        showBanner(NotificationBanner(title: text, subtitle: subtitle, style: .success))
    }

    private func showBanner(_ banner: BaseNotificationBanner){
        //banner.duration = 2.5 <- Won't scroll with this set
        banner.show()
    }

Called as

    successNotificationBanner( * Really long text here * , subtitle: *Subtitle here*)

Also won't scroll doing it the standard way

    let banner = NotificationBanner(title: *etc..*, subtitle: *sub*, style: .success)
    banner.duration = 2.5
    banner.show()

**Comment by Daltron** [comment]
Hi @egenvall! 😄 

So the problem you are running into is actually by design and expected. When the duration is set, the `didSet` method calls this:

```swift
/**
    Updates the scrolling marquee label duration
*/
internal func updateMarqueeLabelsDurations() {
    titleLabel?.speed = .duration(CGFloat(duration - 3))
}
```

The problem is the duration you are setting is not greater than 3, thus causing the label to have a speed duration less than or equal to zero. Based on my findings when I first designed this, I decided to subtract 3 from the duration given because it provides the optimal user experience. We want the banner to stop for a moment before dismissing. With a duration of 2.5 you gave, there simply isn't enough time to scroll all the way to the end and pause momentarily. Let me know if you have any more questions as I will be glad to answer them!

Thanks!

**Comment by egenvall** [comment]
@Daltron 
Thanks for the reply.
Fair enough, I actually skimmed through the code yesterday but I must've been tired ;) 
As a follow up question, if I understand it correctly, the intended behavior is:

- Scroll to the end
- Pause
- Dismiss

During my short experiment my experience was, with the default duration and a long title, it goes like:

- Scroll to end
- Pause
- Scroll back
- Dismiss somewhere when the duration has expired

I'm not sure if that was intended, it made me a bit surprised that it started scrolling back.

Anyway, thank you for a great library.

**Comment by Daltron** [comment]
@egenvall Thanks for pointing this out to me. I see exactly what you are talking about. In the upcoming week, I will try to find a way to make sure that behavior doesn't happen. When I have more info, I will report back!

**Comment by Daltron** [comment]
@egenvall This is still on my todo list and I have not forgotten about it. I'm currently weighing different options and seeing what the best approach might be.

**Comment by Daltron** [comment]
Included in [v1.3.2](https://github.com/Daltron/NotificationBanner/releases/tag/1.3.2) release! 🎉
